### PR TITLE
Refactor methods validate and format of Processo Juridico

### PR DESCRIPTION
**Pull request by pedrovsp**
PR solving issue #28 
Break verification digit validation into a new method.
Change validation logic to use reduce and improve readability.
Add numeric separator to big numbers.
Add new tests.
Add format and validate Processo Juridico to Docs.

### CODE REVIEW

File: docs/pt-br/utilities.md [hunk 1]
```diff
@@ -124,6 +124,22 @@ import { formatCEP } from '@brazilian-utils/brazilian-utils';
 formatCEP('92500000'); // 92500-000
 ```
 
+## isValidProcessoJuridico
+
```
Comment (hyanmandian) [inline docs/pt-br/utilities.md:128]: Missing some description.

File: docs/pt-br/utilities.md [hunk 2]
```diff
@@ -124,6 +124,22 @@ import { formatCEP } from '@brazilian-utils/brazilian-utils';
...
+```javascript
+import { isValidProcessoJuridico } from '@brazilian-utils/brazilian-utils';
+
+isValidProcessoJuridico('00020802520125150049'); // true
+```
+
+## formatProcessoJuridico
+
```
Comment (hyanmandian) [inline docs/pt-br/utilities.md:136]: Missing some description.

File: docs/pt-br/utilities.md (hunk 1 above)
Comment (hyanmandian) [reply docs/pt-br/utilities.md:128]: I think you forgot to add `en` docs too.

File: src/utilities/processo-juridico/index.ts [hunk 3]
```diff
@@ -21,3 +21,34 @@ export function format(processoJuridico: string) {
       return result;
     }, '');
 }
+
+export function verifyDigit(processo: string): boolean {
+  const digits = processo.split('');
+  const verificationDigits = digits.splice(7, 2).join('');
```
Comment (hyanmandian) [inline src/utilities/processo-juridico/index.ts:27]: Maybe you could put `7` and `2` in variables to give some description of what they are

File: src/utilities/processo-juridico/index.ts [hunk 4]
```diff
@@ -21,3 +21,34 @@ export function format(processoJuridico: string) {
...
+
+  const digits1to11 = digits.slice(0, 11).reduce((acc, digit, index) => {
+    return acc + +digit * Math.pow(10, 10 - index);
+  }, 0);
+
+  const firstRemainder = digits1to11 % 97;
```
Comment (hyanmandian) [inline src/utilities/processo-juridico/index.ts:33]: Maybe you could put `97` in a variable to give some description of what it is.

File: src/utilities/processo-juridico/index.ts [hunk 5]
```diff
@@ -21,3 +21,34 @@ export function format(processoJuridico: string) {
...
+
+  const digits12to18 = digits.slice(11).reduce((acc, digit, index) => {
+    return acc + +digit * Math.pow(10, 6 - index);
+  }, 0);
+
+  const secondRemainder = (firstRemainder * 1_000_000_000 + digits12to18 * 100) % 97;
+
+  let verifier = 98 - secondRemainder;
```
Comment (hyanmandian) [inline src/utilities/processo-juridico/index.ts:41]: Why you use `const` for other variables and here you use `let`?

**Review by hyanmandian — CHANGES_REQUESTED** [review]
Hello, @pedrovsp nice job! I requested some changes, please look and tell me if it makes sense to you. Thanks for your contribution!

File: src/utilities/processo-juridico/index.ts (hunk 4 above)
Comment (pedrovsp) [reply src/utilities/processo-juridico/index.ts]: Great suggestion, I added all these numbers into variables and also added a brief comment describing the MOD97_10 formula.

File: src/utilities/processo-juridico/index.ts (hunk 5 above)
Comment (pedrovsp) [reply src/utilities/processo-juridico/index.ts]: Oops, just changed it

File: docs/pt-br/utilities.md (hunk 1 above)
Comment (pedrovsp) [reply docs/pt-br/utilities.md:128]: True, I didn't notice the english docs, added the descriptions on both!

File: src/utilities/processo-juridico/index.ts (hunk 3 above)
Comment (pedrovsp) [reply src/utilities/processo-juridico/index.ts:32]: Sure, also moved them into variables!

**Comment by pedrovsp** [comment]
Hey @hyanmandian thanks for the review, I just pushed changes including all of your suggestions. Let me know if there's anything else that you think it could be different, I'm glad to help.

**Comment by hyanmandian** [comment]
Hi @pedrovsp and sorry for the delay, I was on vacation. Can you resolve the conflicts? After that, I can merge and publish your feature.

Thanks!

**Comment by pedrovsp** [comment]
@hyanmandian , just resolved the conflicts!

**Review by hyanmandian — APPROVED** [review]

**Comment by hyanmandian** [comment]
This function will be published in the V1.

**Pull request merged**